packages/core/src/dialects/abstract/query-interface-typescript.ts

```diff
@@ -686,4 +693,120 @@ export class AbstractQueryInterfaceTypeScript {
   ): Promise<void> {
     await this.sequelize.queryRaw(this.queryGenerator.getToggleForeignKeyChecksQuery(enable), options);
   }
+
+  /**
+   * Create a new savepoint
+   *
+   * @param transaction
+   * @param options
+   */
+  async createSavepoint(transaction: Transaction, options: CreateSavepointOptions): Promise<void> {
+    if (!this.queryGenerator.dialect.supports.savepoints) {
+      throw new Error('Savepoints are not supported for the current dialect!');
+    }
+
+    if (!transaction || !(transaction instanceof Transaction)) {
+      throw new Error('Unable to create a savepoint without transaction object!');
```

Comment:
we could also remove this check now that we're using typescript, but keeping it is fine

```suggestion
      throw new Error('Unable to create a savepoint without the transaction object.');
```

Reply:
I've updated all the error messages